dandiapi/api/views/asset.py

```diff
@@ -440,6 +440,8 @@ def paths(self, request, versions__dandiset__pk: str, versions__version: str, **
             is_folder = folder_index >= 0
 
             if not is_folder:
+                # Ensure base_path is entire filename
+                base_path = asset.path[asset.path.rfind('/') + 1 :]
```

Comment:
could you please add a unittest for this case?

Reply:
Apologies for never providing an update: I attempted to add a quick test for this, but ran into logistical issues. Since at the time I wasn't active on this project, and this enhancement came from work on another repository, I had to abandon this to continue my other obligations. Maybe now that I'm back on the project I can finish this up.

Reply:
Welcome back @AlmightyYakob !

Reply:
@AlmightyYakob please update on adding a unittest -- this is the eldest (6 mo?) PR ATM. We better decide on its destiny asap

Reply:
@AlmightyYakob ping on this PR.

Reply:
Adding a unit test here would be nice, but requires a refactor of our testing setup, and doesn't provide much value, so I'll just merge this.